### PR TITLE
Removed is_error macro dependency (#4098)

### PR DESCRIPTION
**Pull request by awild82**
Replaced depreciated `is_error(ptr)` macro with `ptr == nullptr`.

### CODE REVIEW

File: src/shogun/io/NeuralNetworkFileReader.cpp [hunk 1]
```diff
@@ -51,7 +51,7 @@ CNeuralNetwork* CNeuralNetworkFileReader::read_file(const char* file_path)
 {
 	json_object* json_network = json_object_from_file(file_path);
 
-	if (is_error(json_network))
+	if (json_network == NULL)
```
Comment (vigsterkr) [inline src/shogun/io/NeuralNetworkFileReader.cpp]: since we are c++11 use `nullptr`

File: src/shogun/io/NeuralNetworkFileReader.cpp [hunk 2]
```diff
@@ -68,7 +68,7 @@ CNeuralNetwork* CNeuralNetworkFileReader::read_string(const char* str)
 {
 	json_object* json_network = json_tokener_parse(str);
 
-	if (is_error(json_network))
+	if (json_network == NULL)
```
Comment (vigsterkr) [inline src/shogun/io/NeuralNetworkFileReader.cpp]: `nullptr`

File: src/shogun/io/SerializableJsonFile.cpp [hunk 3]
```diff
@@ -97,7 +95,8 @@ CSerializableJsonFile::init(const char* fname)
 	switch (m_task) {
 	case 'r':
 		buf = json_object_from_file((char*) fname);
-		if (is_error(buf)) {
+		if (buf == NULL)
```
Comment (vigsterkr) [inline src/shogun/io/SerializableJsonFile.cpp]: `nullptr`

File: src/shogun/io/SerializableJsonFile.cpp [hunk 4]
```diff
@@ -219,7 +218,7 @@ CSerializableJsonFile::write_scalar_wrapped(
 		break;
 	}
 
-	if (is_error(m_stack_stream.back()))
+	if (m_stack_stream.back() == NULL)
```
Comment (vigsterkr) [inline src/shogun/io/SerializableJsonFile.cpp]: `nullptr`

File: src/shogun/io/SerializableJsonFile.cpp [hunk 5]
```diff
@@ -289,7 +288,7 @@ CSerializableJsonFile::write_sparse_begin_wrapped(
 	push_object(json_object_new_object());
 
 	json_object* buf = json_object_new_array();
-	if (is_error(buf))
+	if (buf == NULL)
```
Comment (vigsterkr) [inline src/shogun/io/SerializableJsonFile.cpp]: `nullptr`

File: src/shogun/io/SerializableJsonFile.cpp [hunk 6]
```diff
@@ -319,7 +318,7 @@ CSerializableJsonFile::write_sparseentry_begin_wrapped(
 	push_object(buf);
 
 	buf = json_object_new_int(feat_index);
-	if (is_error(buf))
+	if (buf == NULL)
```
Comment (vigsterkr) [inline src/shogun/io/SerializableJsonFile.cpp]: `nullptr`

File: src/shogun/io/SerializableJsonFile.cpp [hunk 7]
```diff
@@ -380,7 +379,7 @@ CSerializableJsonFile::write_sgserializable_begin_wrapped(
 
 	json_object* buf;
 	buf = json_object_new_string(sgserializable_name);
-	if (is_error(buf))
+	if (buf == NULL)
```
Comment (vigsterkr) [inline src/shogun/io/SerializableJsonFile.cpp]: `nullptr`

File: src/shogun/io/SerializableJsonFile.cpp [hunk 8]
```diff
@@ -390,15 +389,15 @@ CSerializableJsonFile::write_sgserializable_begin_wrapped(
 		string_t buf_str;
 		TSGDataType::ptype_to_string(buf_str, generic, STRING_LEN);
 		buf = json_object_new_string(buf_str);
-		if (is_error(buf))
+		if (buf == NULL)
```
Comment (vigsterkr) [inline src/shogun/io/SerializableJsonFile.cpp]: `nullptr`

**Comment by karlnapf** [comment]
@awild82 just asking: this solves the problem in #4098 for you?

**Comment by awild82** [comment]
@karlnapf Yep! I have been successfully compiling and testing with `-DENABLE_JSON=ON`.

**Pull request merged**